samples/nrf9160/gnss/src/assistance.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <zephyr.h>
+#include <modem/modem_info.h>
+#include <modem/modem_jwt.h>
+#include <net/nrf_cloud_rest.h>
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <pm_config.h>
+#include <date_time.h>
+#include <net/nrf_cloud_pgps.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+#include "assistance.h"
+
+static char jwt_buf[600];
+static char rx_buf[2048];
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+static char agps_data_buf[3500];
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+static struct nrf_modem_gnss_agps_data_frame agps_need;
+static struct gps_pgps_request pgps_request;
+static struct nrf_cloud_pgps_prediction *prediction;
+static struct k_work get_pgps_data_work;
+static struct k_work inject_pgps_data_work;
+static K_SEM_DEFINE(time_sem, 0, 1);
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct k_work_q *work_q;
+static volatile bool assistance_active;
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+static int serving_cell_info_get(struct lte_lc_cell *serving_cell)
+{
+	int err;
+
+	err = modem_info_init();
+	if (err) {
+		return err;
+	}
+
+	char resp_buf[MODEM_INFO_MAX_RESPONSE_SIZE];
+
+	err = modem_info_string_get(MODEM_INFO_CELLID,
+				    resp_buf,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		return err;
+	}
+
+	serving_cell->id = strtol(resp_buf, NULL, 16);
+
+	err = modem_info_string_get(MODEM_INFO_AREA_CODE,
+				    resp_buf,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		return err;
+	}
+
+	serving_cell->tac = strtol(resp_buf, NULL, 16);
+
+	/* Request for MODEM_INFO_MNC returns both MNC and MCC in the same string. */
+	err = modem_info_string_get(MODEM_INFO_OPERATOR,
+				    resp_buf,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		return err;
+	}
+
+	serving_cell->mnc = strtol(&resp_buf[3], NULL, 10);
+	/* Null-terminate MCC, read and store it. */
+	resp_buf[3] = '\0';
+	serving_cell->mcc = strtol(resp_buf, NULL, 10);
+
+	return 0;
+}
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+static void get_pgps_data_work_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	int err;
+
+	printk("Sending request for P-GPS predictions to nRF Cloud...\n");
+
+	err = nrf_cloud_jwt_generate(0, jwt_buf, sizeof(jwt_buf));
+	if (err) {
+		printk("Failed to generate JWT, error: %d\n", err);
+
+		return;
+	}
+
+	struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.timeout_ms = CONFIG_NRF_CLOUD_REST_RECV_TIMEOUT * MSEC_PER_SEC,
+		.auth = jwt_buf,
+		.rx_buf = rx_buf,
+		.rx_buf_len = sizeof(rx_buf),
+		.fragment_size = 0, /* use CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE */
```

Comment:
Will it default to using `CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE` when it's set to zero?
If so, the comment could be something like `/* Defaults to CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE when 0 */`